fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Headers.kt

```diff
@@ -116,13 +138,31 @@ class Headers : MutableMap<String, HeaderValues> {
 
     /**
      * Returns the value corresponding to the given [key], or `null` if such a key is not present in the map.
+     *
+     * @param key [String] the header name
+     * @return [HeaderValues] the values at [key] or an empty list
      */
     override fun get(key: String): HeaderValues {
-        return contents[HeaderName(key)].orEmpty()
+        val header = HeaderName(key)
+        return contents[header].orEmpty().let {
+            when (Headers.isSingleValue(header)) {
+                true -> listOf(it.last())
+                false -> it
+            }
```

Comment:
could you consider 
``` kotlin

override fun get(key: String) = contents[HeaderName(key)].orEmpty().let {
    if (Headers.isSingleValue(header)) listOf(it.last()) else it
}
```

Reply:
To me that looks horrible, but if that's our coding style, sure!